src/selection.ts

```diff
@@ -73,8 +74,15 @@ type getFragmentSelection<
   Type extends ObjectLikeType,
   Introspection extends IntrospectionLikeType,
   Fragments extends { [name: string]: any },
+  UnionAccumulator,
 > = Node extends { kind: Kind.INLINE_FRAGMENT; selectionSet: any }
-  ? getSelection<Node['selectionSet']['selections'], Type, Introspection, Fragments>
+  ? getSelection<
+      Node['selectionSet']['selections'],
+      Type,
+      Introspection,
+      Fragments,
+      UnionAccumulator
```

Comment:
Ok, I think I know what confused me about this 😅 Sorry!

The `UnionAccumulator` seems to be an attempt to make this tail-recursive. However, there's no direct recursion going on here.
So, it basically adds indirection, since it hoists this result too much.

Just to double-check this, I did benchmark this and the recursion in the other place does work (intersection), since it's isolated to _one_ recursive type, but `UnionAccumulator` doesn't move the benchmarks and doesn't seem to serve a purpose 🤔